### PR TITLE
fix: disable receive on unsupported chainIds

### LINKED ISSUE #1661: Turn off "Receive" functionality for unsupported IBC Assets
### Overview

Currently, various IBC asset pages are viewable however, the asset is not supported. If an asset is not supported then the "Receive" functionality should be turned off and the button grayed out. If you select "Receive" currently, the receive modal appears but is blank.  (see photos below)

### References and additional details

This affects some IBC asset pages on web

### Acceptance Criteria

A/C: If a specific asset is not supported via the connected wallet the receive functionality should be turned off and the button grayed out. 

### Need By Date

n/a

### Screenshots/Mockups

Screen Shot 2022-05-02 at 4 10 49 PM

Screen Shot 2022-05-02 at 4 49 06 PM

### Estimated effort

_No response_

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Disables receive button for unsupported chainIds

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

closes #1661

## Testing

- Go to the asset page of a `XYZ on Osmosis` asset e.g `Cosmos On Osmosis` (you might have to try that a few times since Imperator API seems to be under heavy load)
- Receive should be disabled

### CODE REVIEW

**Comment by 0xdef1cafe** [comment]
@gomesalexandre let's *not* use `supportedChainIds` from the types package - it doesn't belong there, chain adapters should be the source of truth of supported chain ids, not caip, not our types package.

currently implementation is conceptually ok, but please use `byChainId` rather than `byChain` as we're trying to kill that method and `.chain` property on assets entirely (agnosticism)

File: src/components/AssetHeader/AssetActions.tsx [hunk 1]
```diff
@@ -25,6 +29,15 @@ export const AssetActions: React.FC<AssetActionProps> = ({ assetId, accountId, c
   } = useWallet()
   const asset = useAppSelector(state => selectAssetById(state, assetId))
 
+  useEffect(() => {
+    try {
+      chainAdapterManager.byChain(asset.chain)
```
Comment (0xdef1cafe) [inline src/components/AssetHeader/AssetActions.tsx:34]:
```suggestion
      chainAdapterManager.byChainId(asset.chainId)
```

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]

File: src/components/AssetHeader/AssetActions.tsx [hunk 2]
```diff
@@ -25,6 +29,15 @@ export const AssetActions: React.FC<AssetActionProps> = ({ assetId, accountId, c
   } = useWallet()
   const asset = useAppSelector(state => selectAssetById(state, assetId))
 
+  useEffect(() => {
+    try {
+      chainAdapterManager.byChainId(asset.chainId)
```
Comment (0xApotheosis) [inline src/components/AssetHeader/AssetActions.tsx:34]: `byChainId` returns a `promise` - are we missing an `await`?

**Review by 0xApotheosis — CHANGES_REQUESTED** [review]
One comment. I haven't been able to test yet, as I can't get an Osmosis asset page to load due to server demand.

File: src/components/AssetHeader/AssetActions.tsx (hunk 2 above)
Comment (cjthompson) [reply src/components/AssetHeader/AssetActions.tsx:34]: actually we should change that. `ChainAdapter.getChainID()` is no longer an async function. but we'll need to change that in ChainAdapters first. for now, we need to await that

Comment (cjthompson) [reply src/components/AssetHeader/AssetActions.tsx:34]: See https://github.com/shapeshift/lib/pull/668

Comment (0xApotheosis) [reply src/components/AssetHeader/AssetActions.tsx:34]: I imagine https://github.com/shapeshift/lib/pull/668 will go in pretty quick, so @gomesalexandre it's probably fine to leave this how it is and update this PR to use the latest `chain-adapters` package before we merge it.

Comment (gomesalexandre) [reply src/components/AssetHeader/AssetActions.tsx:34]: Nice one! Moving to draft until #668 is merged, will bump deps and reopen

Comment (0xdef1cafe) [reply src/components/AssetHeader/AssetActions.tsx:34]: gomie @gomesalexandre https://github.com/shapeshift/lib/pull/668 is merged, let's go

Comment (gomesalexandre) [reply src/components/AssetHeader/AssetActions.tsx:34]: Bumped, still unable to test the fix locally once again though as we're still rugged by Imperator.co

**Comment by 0xdef1cafe** [comment]
> One comment. I haven't been able to test yet, as I can't get an Osmosis asset page to load due to server demand.

i think this actually needs a headers csp update

**Comment by gomesalexandre** [comment]
> > One comment. I haven't been able to test yet, as I can't get an Osmosis asset page to load due to server demand.
> 
> i think this actually needs a headers csp update

That will fix the icons but won't unrug Imperator i.e `https://api-osmosis.imperator.co/tokens/v2/<symbol>`

**Review by 0xdef1cafe — APPROVED** [review]

**Review by theoboldfrazier — APPROVED** [review]
Looks good.

**Review by 0xdef1cafe — APPROVED** [review]
confirmed working ty

**Pull request merged**